lighthouse-core/audits/byte-efficiency/offscreen-images.js

```diff
@@ -0,0 +1,146 @@
+      requiredArtifacts: ['ImageUsage', 'ViewportDimensions', 'traces', 'networkRecords']
+    };
+  }
+
+  /**
+   * @param {!ClientRect} imageRect
+   * @param {{innerWidth: number, innerHeight: number}} viewportDimensions
+   * @return {number}
+   */
+  static computeVisiblePixels(imageRect, viewportDimensions) {
+    const innerWidth = viewportDimensions.innerWidth;
+    const innerHeight = viewportDimensions.innerHeight;
+
+    const top = Math.max(imageRect.top, -1 * ALLOWABLE_OFFSCREEN_Y);
+    const right = Math.min(imageRect.right, innerWidth + ALLOWABLE_OFFSCREEN_X);
+    const bottom = Math.min(imageRect.bottom, innerHeight + ALLOWABLE_OFFSCREEN_Y);
+    const left = Math.max(imageRect.left, -1 * ALLOWABLE_OFFSCREEN_X);
+
+    return Math.max(right - left, 0) * Math.max(bottom - top, 0);
+  }
+
+  /**
+   * @param {!Object} image
+   * @param {{innerWidth: number, innerHeight: number}} viewportDimensions
+   * @return {?Object}
+   */
+  static computeWaste(image, viewportDimensions) {
+    const url = URL.getDisplayName(image.src, {preserveQuery: true});
+    const totalPixels = image.clientWidth * image.clientHeight;
+    const visiblePixels = this.computeVisiblePixels(image.clientRect, viewportDimensions);
+    const wastedRatio = 1 - visiblePixels / totalPixels;
+    const totalBytes = image.networkRecord.resourceSize;
+    const wastedBytes = Math.round(totalBytes * wastedRatio);
+
+    if (!Number.isFinite(wastedRatio)) {
+      return new Error(`Invalid image sizing information ${url}`);
+    }
+
+    return {
+      url,
+      preview: {
+        url: image.networkRecord.url,
+        mimeType: image.networkRecord.mimeType
+      },
+      requestStartTime: image.networkRecord.startTime,
+      totalBytes,
+      wastedBytes,
+      wastedPercent: 100 * wastedRatio,
+    };
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {{results: !Array<Object>, tableHeadings: Object,
+   *     passes: boolean=, debugString: string=}}
+   */
+  static audit_(artifacts) {
+    const images = artifacts.ImageUsage;
+    const viewportDimensions = artifacts.ViewportDimensions;
+
+    let debugString;
+    const resultsMap = images.reduce((results, image) => {
+      if (!image.networkRecord) {
+        return results;
+      }
+
+      const processed = OffscreenImages.computeWaste(image, viewportDimensions);
+      if (processed instanceof Error) {
+        debugString = processed.message;
+        return results;
+      }
+
+      // If an image was used more than once, warn only about its least wasteful usage
+      const existing = results.get(processed.preview.url);
+      if (!existing || existing.wastedBytes > processed.wastedBytes) {
+        results.set(processed.preview.url, processed);
+      }
+
+      return results;
+    }, new Map());
+
+    return TTIAudit.audit(artifacts).then(ttiResult => {
+      const tti = ttiResult.extendedInfo.value.timestamps.timeToInteractive / 1000000;
```

Comment:
`ttiInSeconds` or something like that? Any reason you aren't just using `ttiResult.rawValue`?